### PR TITLE
[TorchMhlo] add slices

### PR DESCRIPTION
**Pull request by fortianyou**
- Sort the TorchMLIR white list
- [TorchMhlo] add slices

### CODE REVIEW

**Review by wyzero — APPROVED** [review]
LGTM

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 1]
```diff
@@ -1326,6 +1329,76 @@ class ConvertAtenLinearOp : public ConvertAtenMatmulBaseOp<AtenOpT> {
...
+    return rewriter.notifyMatchFailure(
+        op, "Only constant dim is currently supported");
+
+  auto rank = selfTy.getRank();
+  if (rank == 0) {
+    return rewriter.notifyMatchFailure(
+        op, "The rank of tensor must be greater than 0");
+  }
+
+  dim = (dim + rank) % rank;
```
Comment (qiuxiafei) [inline pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1352]: `torch.squeeze` also support `dim=None` in which case all size-1 dims are removed. But it seems that we only sequeeze on explicity size-1 dim ?

File: pytorch_blade/tests/mhlo/slices.mlir [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// CHECK-SAME:         %[[ARG0:.*]]: tensor<4x65x256xf32>) -> tensor<4x33x256xf32> {
+// CHECK:         %[[T0:.*]] = "mhlo.slice"(%[[ARG0]]) {limit_indices = dense<[4, 65, 256]> : tensor<3xi64>, start_indices = dense<0> : tensor<3xi64>, strides = dense<[1, 2, 1]> : tensor<3xi64>} : (tensor<4x65x256xf32>) -> tensor<4x33x256xf32>
+// CHECK:         return %[[T0]] : tensor<4x33x256xf32>
+func @torch.aten.slice.none(%arg0: !torch.vtensor<[4,65,256],f32>) -> !torch.vtensor<[4,33,256],f32> {
+  %int1 = torch.constant.int 1
+  %int2 = torch.constant.int 2
+  %none = torch.constant.none
+  %0 = torch.aten.slice.Tensor %arg0, %int1, %none, %none, %int2 : !torch.vtensor<[4,65,256],f32>, !torch.int, !torch.none, !torch.none, !torch.int -> !torch.vtensor<[4,33,256],f32>
+  return %0 : !torch.vtensor<[4,33,256],f32>
+}
```
Comment (qiuxiafei) [inline pytorch_blade/tests/mhlo/slices.mlir:48]: also filechecking on torch.squeeze?

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp (hunk 1 above)
Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:1352]: The `AtenSqueezeDimOp` here only means squeeze the specified dimension.

File: pytorch_blade/tests/mhlo/slices.mlir (hunk 2 above)
Comment (fortianyou) [reply pytorch_blade/tests/mhlo/slices.mlir:48]: I will add squeeze tests in another PR.

**Pull request merged**